frontend/src/components/Drawer.js

```diff
@@ -0,0 +1,133 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import PropTypes from 'prop-types';
+import './Drawer.scss';
+import useOnClickOutside from '../hooks/useOnOutsideClick';
+
+const ESCAPE_KEY_CODE = 27;
+
+export default function Drawer({
+  title,
+  footer,
+  stickyHeader,
+  stickyFooter,
+  children,
+  triggerRef,
+}) {
+  const [isOpen, setIsOpen] = useState(false);
+  const elementRef = useRef(null);
+  const closeButtonRef = useRef(null);
+
+  const headerHeight = useMemo(() => {
```

Comment:
Any big diff between **memo** with empty dependency array vs **useEffect** with an empty array?

Reply:
with `useMemo` I don't need to also manager header height state in some `useState` state container. instead, I can just use the return value, simplifying things a bit.